corehq/apps/app_manager/models.py

```diff
@@ -2173,15 +2177,16 @@ class CaseSearchProperty(DocumentSchema):
     Case properties available to search on.
     """
     name = StringProperty()
-    label = DictProperty()
+    label = LabelProperty()
     appearance = StringProperty(exclude_if_none=True)
     input_ = StringProperty(exclude_if_none=True)
     default_value = StringProperty(exclude_if_none=True)
-    hint = DictProperty()
+    hint = LabelProperty()
     hidden = BooleanProperty(default=False)
     allow_blank_value = BooleanProperty(default=False)
     exclude = BooleanProperty(default=False)
     required = StringProperty(exclude_if_none=True)
+    required_message = LabelProperty()
     validation = SchemaListProperty(CaseSearchValidationCondition)
```

Comment:
I'm tempted to do this:
```suggestion
    required = SchemaProperty(CustomAssertion)
    validation = SchemaListProperty(CustomAssertion)
```
But that'd require a (lazy) migration, and I'm not totally sure it's clearer.  They are conceptually assertions, sort of, and are communicated the same way in the suite XML

Reply:
I think it'd be a little clearer, but defer to you on whether or not a migration would be worth it. We've done a bunch of case search property config migrations - maybe you've already done one, but if you haven't, the one deleted in this PR is a good example: https://github.com/dimagi/commcare-hq/pull/31771/files

Reply:
Since you've changed the suite model to use the same XML class I think it makes sense to do the same thing here. Depending on the timing you could piggy back on the _entered convention migration.

Reply:
I spent a frustrating amount of time on this today.  The chief issue I ran into was that jsonobject doesn't seem to respect `required=False` on `SchemaProperty`s.  This is minimally reproducible with
```python
from jsonobject import *

class Assertion(JsonObject):
    test = StringProperty(required=True)

class SearchConfig(JsonObject):
    required = ObjectProperty(Assertion, required=False)

SearchConfig.wrap({})  # raises BadValueError: Property test is required.
```
This effectively makes `required` required, which it shouldn't be.  I can think of two workarounds for this:
* Make another `Assertion` class where `required=False` and use that here
* Make `required` a `SchemaListProperty` like `validations`, signifying that it can contain 0 or 1 (or any other number)

I ended up implementing the first of those options, but I'm not in love with it.  :shrug: